### PR TITLE
Update signature on unstable_scheduleCallback

### PR DESCRIPTION
**Pull request by mauricedb**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: See the current signature here: https://github.com/facebook/react/blob/master/packages/scheduler/src/Scheduler.js#L295
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] ~~If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.~~

### CODE REVIEW

File: types/scheduler/index.d.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const unstable_NormalPriority = 3;
 export const unstable_IdlePriority = 5;
 export const unstable_LowPriority = 4;
 export function unstable_runWithPriority<T>(priorityLevel: number, eventHandler: () => T): T | undefined;
-export function unstable_scheduleCallback(callback: FrameCallbackType, deprecated_options?: { timeout: number}): CallbackNode;
+export function unstable_scheduleCallback(priorityLevel: number, callback: FrameCallbackType, options?: { delay: number}): CallbackNode;
```
Comment (eps1lon) [inline types/scheduler/index.d.ts]: Timeout is still an option: https://github.com/facebook/react/blob/d346d92084316ec10714b9ac4bfb1ebcfdf05fb9/packages/scheduler/src/Scheduler.js#L308 and both of them are optional

**Review by eps1lon — CHANGES_REQUESTED** [review]
Thank you for updating them.

Please don't forget to to update the version header to 0.16

File: types/scheduler/index.d.ts (hunk 1 above)
Comment (mauricedb) [reply types/scheduler/index.d.ts]: @eps1lon Oops, missed that one. The timeout back in now and both timeout and delay are optional.

File: types/scheduler/test/scheduler.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
-import { unstable_scheduleCallback, unstable_cancelCallback, unstable_now } from 'scheduler';
+import { unstable_scheduleCallback, unstable_cancelCallback, unstable_now, unstable_NormalPriority } from 'scheduler';
 
 // $ExpectType CallbackNode
-const callbackNode = unstable_scheduleCallback(() => {}, { timeout: 100 });
+const callbackNode = unstable_scheduleCallback(unstable_NormalPriority, () => {}, { delay: 100 });
```
Comment (eps1lon) [inline types/scheduler/test/scheduler.ts]: Last nitpick: Only adjust this test to pass and add a new one for delay.

**Review by eps1lon — APPROVED** [review]

**Pull request merged**